src/app/server/Server.cpp

```diff
@@ -227,10 +231,33 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
     return err;
 }
 
+CHIP_ERROR Server::InitCASESessionManager()
+{
+    ReturnErrorOnFailure(mDNSResolver.Init(DeviceLayer::UDPEndPointManager()));
+    DeviceProxyInitParams initParams = {
+        .sessionManager = &mSessions,
+        .exchangeMgr    = &mExchangeMgr,
+        .idAllocator    = &mSessionIDAllocator,
+        .fabricTable    = &mFabrics,
+        .clientPool     = &mCASEClientPool,
+        .imDelegate     = nullptr,
+    };
+    CASESessionManagerConfig sessionManagerConfig = {
+        .sessionInitParams = initParams,
+        .dnsCache          = nullptr,
+        .devicePool        = &mDevicePool,
+        .dnsResolver       = &mDNSResolver,
+    };
+    mCASESessionManager = new (&mCASESessionManagerStorage) CASESessionManager(sessionManagerConfig);
+
+    return mCASESessionManager == nullptr ? CHIP_ERROR_NO_MEMORY : CHIP_NO_ERROR;
+}
+
 void Server::Shutdown()
 {
     chip::Dnssd::ServiceAdvertiser::Instance().Shutdown();
     chip::app::InteractionModelEngine::GetInstance()->Shutdown();
+    mCASESessionManager->~CASESessionManager();
```

Comment:
This will end up running the destructor again ona destructed object when ~Server runs.  Does that just never happen?

Reply:
The `mCASESessionManager` is a pointer so when `~Server` runs the descructor won't be called.

Reply:
How about allocated memory? How is mCASESessionManager lifetime managed?

I find calling destructors directly when not implementing smart pointers to be a potential code smell. Why do we use destructor and not a 'cleanup' method in case we actually want 'cleanup without memory free'?

Reply:
The `mCASESessionManager` is allocated in an `aligned_storage_t`. It's mainly there because `CASESessionManager` does all the initialization in the constructor while `Server` class defers it to the `Initialize` function. I have to use `new` to initialize the `CASESessionManager`.

I chose to use in-place new rather than allocating dynamic memory for more derministric behaivor. For in-place new calling destructor directly replaces `delete`.

Reply:
Splitted the `CASESessionManager` initialization logic. Now `mCASESessionManager` is a member object.